src/main/java/lotto/domain/WinningRecord.java

```diff
@@ -1,40 +1,31 @@
 package lotto.domain;
 
-public class WinningRecord {
+public final class WinningRecord {
 
     private final Lotto winningLotto;
 
-    private final WinningCount winningCount;
+    private final LottoNumber bonusBall;
 
-    public WinningRecord(Lotto winningLotto) {
-        this.winningLotto = winningLotto;
-        this.winningCount = new WinningCount();
+    public WinningRecord(Lotto winningLotto, int bonusBall) {
+        this(winningLotto, new LottoNumber(bonusBall));
     }
 
-    public void record(Lottoes lottoes) {
-        for (Lotto lotto : lottoes.getLottoes()) {
-            int matchCount = countMatchingNumbers(lotto);
-            winningCount.increaseCount(matchCount);
-        }
+    public WinningRecord(Lotto winningLotto, LottoNumber bonusBall) {
+        this.winningLotto = winningLotto;
+        this.bonusBall = bonusBall;
     }
 
-    private int countMatchingNumbers(Lotto lotto) {
-        return (int) lotto.getNumbers()
-                .stream()
-                .filter(number -> winningLotto.getNumbers().contains(number))
-                .count();
-    }
+    public WinningCount record(final Lottoes lottoes) {
+        final WinningCount winningCount = new WinningCount();
 
-    public int getMatchCount(int matchCount) {
-        return winningCount.getMatchCount(matchCount);
-    }
+        for (Lotto lotto : lottoes.getLottoes()) {
+            int matchCount = winningLotto.countContainNumbers(lotto);
+            boolean matchBonus = lotto.contains(bonusBall);
+            Rank rank = Rank.valueOf(matchCount, matchBonus);
```

Comment:
```suggestion
        Map<Rank, Long> collect = lottoes.getLottoes()
            .stream()
            .map(lotto -> Rank.valueOf(winningLotto.countContainNumbers(lotto), lotto.contains(bonusBall)))
            .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
```

winningCount는 WinningRecord에서 생성된 이후 내부의 값은 불변인 편이 좋을 것 같은데요,
현재 구조에서는 아쉽게도 increaseCount를 통해 언제든지 변경이 가능한 구조네요 😊

코멘트 드린 방법처럼 Map을 만들어서 winningCount 생성자에 바로 넘기고, 
winningCount 내부에서는 unmodifiableMap으로 가지고 있는다면 불변이 보장되는 좀 더 안전한 코드가 될 것 같습니다 :)